docs/source/guide/minibatch-parallelism.rst

```diff
@@ -0,0 +1,55 @@
+.. _guide-minibatch-parallelism:
+
+6.8 Data Loading Parallelism
+-----------------------
+
+In minibatch training of GNNs, we usually need to cover several stages to
+generate a minibatch, including:
+
+* Iterate over item set and generate minibatch seeds in batch size.
+* Sample negative items for each seed from graph.
+* Sample neighbors for each seed from graph.
+* Exclude seed edges from the sampled subgraphs.
+* Fetch node and edge features for the sampled subgraphs.
+* Convert the sampled subgraphs to DGLMiniBatches.
+* Copy the DGLMiniBatches to the target device.
+
+.. code:: python
+
+    datapipe = gb.ItemSampler(itemset, batch_size=1024, shuffle=True)
+    datapipe = datapipe.sample_uniform_negative(g, 5)
+    datapipe = datapipe.sample_neighbor(g, [10, 10]) # 2 layers.
+    datapipe = datapipe.transform(gb.exclude_seed_edges)
+    datapipe = datapipe.fetch_feature(feature, node_feature_keys=["feat"])
+    datapipe = datapipe.to_dgl()
+    datapipe = datapipe.copy_to(device)
+    dataloader = gb.MultiProcessDataLoader(datapipe, num_workers=0)
+
+All these stages are implemented in separate
+`IterableDataPipe <https://pytorch.org/data/main/torchdata.datapipes.iter.html>`__
+and stacked together with `PyTorch DataLoader <https://pytorch.org/docs/stable/data
+.html#torch.utils.data.DataLoader>`__.
+This design allows us to easily customize the data loading process by
+chaining different data pipes together. For example, if we want to sample
+negative items for each seed from graph, we can simply chain the
+:class:`~dgl.graphbolt.NegativeSampler` after the :class:`~dgl.graphbolt.ItemSampler`.
+
+But simply chaining data pipes together incurs performance overheads. Considering
+the fact that some stages are computationally intensive while others are I/O intensive,
```

Comment:
instead mentioning computationally intensive and I/O intensive, probably better to say they are using different hardware resources which includes CPU intensive, GPU intensive, PCIe (CPU to GPU), Disk e.t.c.

Reply:
updated.